src/cmd/key/renew.sh

```diff
@@ -28,11 +28,14 @@ Try first:  $(basename $0) key join
     local today=$(date -d $(date +%F) +%s)
     time=$(( ( $expday - $today ) / 86400 ))
 
-    local commands=";expire;$time;y;key 1;expire;$time;y;key 1;save"
-    commands=$(echo "$commands" | tr ';' "\n")
-    echo -e "$commands" | gpg --no-tty --command-fd=0 --key-edit $GPG_KEY 2>/dev/null
+    # blank quotes ("") => None
+    local commands=("" "expire" "" "$time" "" "key 1" "" "expire" "" "$time" "" "save" "" "")
+    call_gpg fn/interact.py "$GPG_KEY" "${commands[@]}"
```

Comment:
The same comment as before here. The only argument passed here should be the `$time` of expiration (besides the action, which may be `set-expiration-time`).
Even better, you can give the `$expdate` as an argument and let the python do the calculations.